corehq/apps/hqcase/case_helper.py

```diff
@@ -1,293 +1,188 @@
-from couchdbkit import ResourceNotFound
-from jsonschema import validate
-
-from casexml.apps.case.mock import CaseBlock, IndexAttrs
+from dataclasses import dataclass
 
 from corehq.apps.users.models import CouchUser
 
-from .utils import SYSTEM_FORM_XMLNS, submit_case_blocks
-
-case_block_args_case_properties = {
-    'owner_id',
-    'external_id',
-    'case_type',
-    'case_name',
-}
+from .api.updates import handle_case_update
 
 
 class CaseHelper:
     """
     CaseHelper aims to offer a simple interface for simple operations on
     cases.
 
-    Initialize ``CaseHelper`` with an existing case::
+    Initialize ``CaseHelper`` with an existing case ... ::
+
+        domain = 'hawaiian-mythology'
+        helper = CaseHelper(case=case, domain=domain)
 
-        helper = CaseHelper(case)
+    ... or with a case ID::
 
-    Or create a case using data in the format returned by
-    ``CommCareCase.to_api_json(lite=True)``::
+        helper = CaseHelper(case_id=case_id, domain=domain)
 
-        helper = CaseHelper()
+    Or create a case using data in the format returned by the Case API
+    v0.6 ``serialize_case()`` function::
+
+        helper = CaseHelper(domain=domain)
         helper.create_case({
-            'case_id': uuid4().hex,
             'domain': 'hawaiian-mythology',
-            'properties': {
-                'case_name': 'Namaka',
-                'case_type': 'child',
-            },
+            'case_type': 'child',
+            'case_name': 'Namaka',
         })
 
-    Update the case::
+    Update case properties::
 
-        helper.update(properties={'description': 'Goddess of the sea'})
+        helper.update({'properties': {
+            'description': 'Goddess of the sea',
+        }})
 
     Add an index::
 
-        helper.update(indices={'mother': {
+        helper.update({'indices': {'mother': {
             'case_id': mother.case_id,
             'case_type': mother.type,
             'relationship': 'child',
-        }})
+        }}})
 
     Close the case::
 
         helper.close()
 
     ``create_case()``, ``update()``, and ``close()`` methods all accept
-    ``user_id``, ``device_id`` and ``xmlns`` parameters to identify
-    who/what called the operation. Under the hood, all operations use
-    ``submit_case_blocks()`` to submit forms.
+    ``user_id`` and ``device_id`` parameters to identify who/what called
+    the operation. Under the hood, all operations use the Case API v0.6
+    ``handle_case_update()`` function.
 
     """
 
-    def __init__(self, case=None):
-        self.case = case
+    def __init__(self, *, domain, case=None, case_id=None):
+
+        @dataclass
+        class BriefCase:
+            """Like a CommCareCase, but briefer."""
```

Comment:
:briefcase:

Reply:
(Yeah, I have got to admit, that class name is my favorite part. :joy: )